snippets/firestore/firestore_pipelines.py

```diff
@@ -0,0 +1,1289 @@
+                .as_("isMysteryOrFantasy")
+        ) \
+        .execute()
+    # [END array_contains_any]
+    for res in result:
+        print(res)
+
+def array_length():
+    # [START array_length]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("genre").array_length().as_("genreCount")) \
+        .execute()
+    # [END array_length]
+    for res in result:
+        print(res)
+
+def array_reverse():
+    # [START array_reverse]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("genre").array_reverse().as_("reversedGenres")) \
+        .execute()
+    # [END array_reverse]
+    for res in result:
+        print(res)
+
+def equal_function():
+    # [START equal_function]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("rating").equal(5).as_("hasPerfectRating")) \
+        .execute()
+    # [END equal_function]
+    for res in result:
+        print(res)
+
+def greater_than_function():
+    # [START greater_than]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("rating").greater_than(4).as_("hasHighRating")) \
+        .execute()
+    # [END greater_than]
+    for res in result:
+        print(res)
+
+def greater_than_or_equal_to_function():
+    # [START greater_or_equal]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("published").greater_than_or_equal(1900).as_("publishedIn20thCentury")) \
+        .execute()
+    # [END greater_or_equal]
+    for res in result:
+        print(res)
+
+def less_than_function():
+    # [START less_than]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("published").less_than(1923).as_("isPublicDomainProbably")) \
+        .execute()
+    # [END less_than]
+    for res in result:
+        print(res)
+
+def less_than_or_equal_to_function():
+    # [START less_or_equal]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("rating").less_than_or_equal(2).as_("hasBadRating")) \
+        .execute()
+    # [END less_or_equal]
+    for res in result:
+        print(res)
+
+def not_equal_function():
+    # [START not_equal]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("title").not_equal("1984").as_("not1984")) \
+        .execute()
+    # [END not_equal]
+    for res in result:
+        print(res)
+
+def exists_function():
+    # [START exists_function]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(Field.of("rating").exists().as_("hasRating")) \
+        .execute()
+    # [END exists_function]
+    for res in result:
+        print(res)
+
+def and_function():
+    # [START and_function]
+    result = client.pipeline() \
+        .collection("books") \
+        .select(
+            And(
```

Comment:
Maybe you should include the import in the sample? `from google.cloud.firestore_v1.pipeline_expressions import And`. I guess this feedback would apply to Field and Constant too though...

Our usual policy was that samples should all include relevant imports. Most of these pipeline tests don't really need imports. But when we start adding Ands and Ors and Nots, I could see users getting confused, since they look so close to built-in methods